extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -187,12 +195,32 @@ private function getThemeName( $params, $value ) {
 	}
 
 	private function getLayout( $params ) {
+		$validator = new InfoboxParamsValidator();
 		$layoutName = isset( $params[ 'layout' ] ) ? $params[ 'layout' ] : false;
-		if ( $layoutName && in_array( $layoutName, $this->supportedLayouts ) ) {
+		if ( $validator->validateLayout( $layoutName ) ) {
 			//make sure no whitespaces, prevents side effects
 			return self::INFOBOX_LAYOUT_PREFIX . $layoutName;
 		}
 
 		return self::INFOBOX_LAYOUT_PREFIX . self::DEFAULT_LAYOUT_NAME;
 	}
+
+	private function getColor( $colorName, $params, PPFrame $frame ) {
+		$validator = new InfoboxParamsValidator();
+
+		$sourceParam = $colorName . '-source';
+		$defaultParam = $colorName . '-default';
+
+		if ( isset( $params[ $sourceParam ] ) && !empty( $frame->getArgument( $params[ $sourceParam ] ) ) ) {
```

Comment:
Spacing is wrong, see https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#array-syntax